services/datamanager/server_test.go

```diff
@@ -83,3 +85,27 @@ func TestServerSync(t *testing.T) {
 		})
 	}
 }
+
+func TestServerDoCommand(t *testing.T) {
+	resourceMap := map[resource.Name]interface{}{
+		datamanager.Named(testDataManagerServiceName): &inject.DataManagerService{
+			DoCommandFunc: generic.EchoFunc,
+		},
+	}
+	server, err := newServer(resourceMap)
+	test.That(t, err, test.ShouldBeNil)
+
+	cmd, err := protoutils.StructToStructPb(generic.TestCommand)
+	test.That(t, err, test.ShouldBeNil)
+	doCommandRequest := &commonpb.DoCommandRequest{
+		Name:    testDataManagerServiceName,
+		Command: cmd,
+	}
+	doCommandResponse, err := server.DoCommand(context.Background(), doCommandRequest)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Assert that do command response is an echoed request.
+	respMap := doCommandResponse.Result.AsMap()
+	test.That(t, respMap["command"], test.ShouldResemble, "test")
```

Comment:
Annoyingly, `test.That(t, resp, test.ShouldResemble, generic.TestCommand)` fails because somewhere the `int` 500 is converted to a `float64`.